src/common/args.h

```diff
@@ -85,27 +85,99 @@ struct SectionInfo {
 };
 
 std::string SettingToString(const common::SettingsValue&, const std::string&);
-std::optional<std::string> SettingToString(const common::SettingsValue&);
-
 int64_t SettingToInt(const common::SettingsValue&, int64_t);
-std::optional<int64_t> SettingToInt(const common::SettingsValue&);
-
 bool SettingToBool(const common::SettingsValue&, bool);
-std::optional<bool> SettingToBool(const common::SettingsValue&);
 
 class ArgsManager
 {
 public:
     /**
-     * Flags controlling how config and command line arguments are validated and
-     * interpreted.
+     * Flags controlling how config and command line arguments are parsed.
+     *
+     * The flags below provide very basic type checking, designed to catch
+     * obvious configuration mistakes and provide helpful error messages.
+     * Specifying these flags is not a substitute for actually validating
+     * setting values that are parsed and making sure they are legitimate.
+     *
+     * Summary of recommended flags:
+     *
+     * - For most settings, just use standalone ALLOW_BOOL, ALLOW_INT, or
+     *   ALLOW_STRING flags.
+     *
+     * - If your setting accepts multiple values and you want to read all the
+     *   values, not just the last value, add | ALLOW_LIST to the flags.
+     *
+     * - If your setting is a command that causes a new action to be performed,
+     *   and does not require a value, add | ALLOW_BOOL to the flags. Adding it
+     *   just allows the setting to be specified alone on the command line
+     *   without a value, as "-foo" instead of "-foo=value".
+     *
+     * - Only use the DISALLOW_NEGATION flag if your setting really cannot
+     *   function without a value, so the command line interface will generally
+     *   support negation and be more consistent.
+     *
+     * Detailed description of flags:
+     *
+     * The ALLOW_STRING, ALLOW_INT, and ALLOW_BOOL flags control what syntaxes are
+     * accepted, according to the following chart:
+     *
+     * | Syntax   | STRING | INT | BOOL | STRING|BOOL | INT|BOOL |
+     * | -------- | ------ | --- | ---- | ----------- | -------- |
+     * | -foo=abc |   X    |     |      |      X      |          |
+     * | -foo=123 |   X    |  X  |      |      X      |    X     |
+     * | -foo=0   |   X    |  X  |  X   |      X      |    X     |
+     * | -foo=1   |   X    |  X  |  X   |      X      |    X     |
+     * | -foo     |        |     |  X   |      X      |    X     |
+     * | -foo=    |   X    |  X  |  X   |      X      |    X     |
+     * | -nofoo   |   X    |  X  |  X   |      X      |    X     |
+     * | -nofoo=1 |   X    |  X  |  X   |      X      |    X     |
```

Comment:
Possible fixes for Doxygen issues I found:
* Table generation failure due to `|` in headers. (Tried to surround headers with markdown backticks but couldn't get it working that way).
* Code blocks not being generated due to insufficient indentation.
* File reference not becoming a link.
```diff
diff --git a/src/common/args.h b/src/common/args.h
index d36b491d0b..23ced9e3e3 100644
--- a/src/common/args.h
+++ b/src/common/args.h
@@ -121,16 +121,16 @@ public:
      * The ALLOW_STRING, ALLOW_INT, and ALLOW_BOOL flags control what syntaxes are
      * accepted, according to the following chart:
      *
-     * | Syntax   | STRING | INT | BOOL | STRING|BOOL | INT|BOOL |
-     * | -------- | ------ | --- | ---- | ----------- | -------- |
-     * | -foo=abc |   X    |     |      |      X      |          |
-     * | -foo=123 |   X    |  X  |      |      X      |    X     |
-     * | -foo=0   |   X    |  X  |  X   |      X      |    X     |
-     * | -foo=1   |   X    |  X  |  X   |      X      |    X     |
-     * | -foo     |        |     |  X   |      X      |    X     |
-     * | -foo=    |   X    |  X  |  X   |      X      |    X     |
-     * | -nofoo   |   X    |  X  |  X   |      X      |    X     |
-     * | -nofoo=1 |   X    |  X  |  X   |      X      |    X     |
+     * | Syntax   | STRING | INT | BOOL | STRING\|BOOL | INT\|BOOL |
+     * | -------- | :----: | :-: | :--: | :----------: | :-------: |
+     * | -foo=abc |   X    |     |      |      X       |           |
+     * | -foo=123 |   X    |  X  |      |      X       |     X     |
+     * | -foo=0   |   X    |  X  |  X   |      X       |     X     |
+     * | -foo=1   |   X    |  X  |  X   |      X       |     X     |
+     * | -foo     |        |     |  X   |      X       |     X     |
+     * | -foo=    |   X    |  X  |  X   |      X       |     X     |
+     * | -nofoo   |   X    |  X  |  X   |      X       |     X     |
+     * | -nofoo=1 |   X    |  X  |  X   |      X       |     X     |
      *
      * Once validated, settings can be retrieved by called GetSetting(),
      * GetArg(), GetIntArg(), and GetBoolArg(). GetSetting() is the most general
@@ -234,10 +234,10 @@ protected:
      *
      * Examples:
      *
-     *   GetArg("-foo") returns "abc" if -foo=abc was specified, or nullopt if unset
-     *   GetIntArg("-foo") returns 123 if -foo=123 was specified, or nullopt if unset
-     *   GetBoolArg("-foo") returns true if -foo was specified, or nullopt if unset
-     *   GetBoolArg("-foo") returns false if -nofoo was specified, or nullopt if unset
+     *     GetArg("-foo")     // returns "abc" if -foo=abc was specified, or nullopt if unset
+     *     GetIntArg("-foo")  // returns 123 if -foo=123 was specified, or nullopt if unset
+     *     GetBoolArg("-foo") // returns true if -foo was specified, or nullopt if unset
+     *     GetBoolArg("-foo") // returns false if -nofoo was specified, or nullopt if unset
      *
      * If no type flags (ALLOW_STRING, ALLOW_INT, or ALLOW_BOO* Table generation failure due to `|` in headersL) are set, GetArg
      * functions do many type coercions and can have surprising behaviors which
@@ -263,17 +263,17 @@ protected:
      *   handle boolean settings.json or command line values (-foo -nofoo)
      *   differently than string values (-foo=abc), you can write:
      *
-     *     if (auto foo{args.GetBoolArg("-foo")}) {
-     *         // handle bool foo.value()
-     *     } else if (auto foo{args.GetArg("-foo")}) {
-     *         // handle string foo.value()
-     *     } else {
-     *         // handle unset setting
-     *     }
+     *       if (auto foo{args.GetBoolArg("-foo")}) {
+     *           // handle bool foo.value()
+     *       } else if (auto foo{args.GetArg("-foo")}) {
+     *           // handle string foo.value()
+     *       } else {
+     *           // handle unset setting
+     *       }
      *
      * More examples of GetArg function usage can be found in the
      * @ref example_options::ReadOptions() function in
-     * @ref ../test/argsman_tests.cpp
+     * @ref argsman_tests.cpp
      *@{*/
     std::optional<std::string> GetArg(const std::string& strArg) const;
     std::optional<int64_t> GetIntArg(const std::string& strArg) const;
```

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16545#discussion_r1699171640

Thank you! I should try building the docs myself, but I did apply your patch and added a @name to try to fix the grouping